### DISCUSSION TITLE
Choose email address for merge commit on Web UI when merging

### DISCUSSION THREAD

**Discussion by ndrewtl** [community/community#27375] [Pull Requests]
Currently, when merging a PR, the merge commit associated with that PR is the automatically set to the user's primary email address, or the default noreply email address if their emails are set to private. However, for those not wishing to use Github in different contexts, this may be difficult. For instance, if a user works for an organization and has an organization repo, then their merge commit email address should be their company email address, and should not expose their private or personal email address at commit time.

There should also be a difference between default primary emails and the email address used for commits. For instance, a user might have a primary email that should be used for administration of their GitHub account, notifications, etc, but not want that email to be associated with commits.

All in all, what I would like here is a level of granularity in how GitHub handles email addresses. Being able to set the email used for commits on a per-repository basis would be ideal, but in the meantime, I think the simplest, easiest solution is just to offer the user a select box asking which email to use for the commit when a merge happens.

**Comment by geoffh-etched** [comment]
Worse: if you are enforcing merge rules to ensure use of the correct (company) email, then this rewrite will break those rules and prevent merging!

**Comment by ChrisIdema** [comment]
Still not fixed after more than 3 years.
I have 2 verified email addresses for my github account. 1 I use for notifications. This is my private email. The other is my public email. It's a separate email account I only use for commits.

If I check "Keep my email addresses private" github inserts a weird email address. I used this in the past. But this is not a valid address and is not accepted everywhere. So I added this second address. I configured this address on my PC as the default author email. Unfortunately github chooses my private email address as commit email if I let it create an initial commit with a readme. It did so without my consent. Fortunately the repo is still private and I was able to rewrite history and force push the changes. But this is a major privacy risk.

**Comment by bliu-ccsf** [comment]
This is low-key a huge problem especially if the PR gets merged with an email that wasn't intended. We should be allowed to select a preferred email for each organization and that would help solve a good chunk of this problem. This issue leaves people open to being data mined and personal emails spammed, people being reprimanded for mistakenly using the wrong email in an organization, or cause an organization to fail security audits.

I also encountered this:
```text
The "Verified Domain" Strategy

If you are an admin for the organization, you can enable Email Domain Restrictions. While this doesn't automatically "select" the right email for the user, it will block the merge if the user tries to commit with a personal email that doesn't match the organization's domain. It’s a "hard" way to force users to switch their email in the dropdown before finishing the merge.

Since there isn't an automated "Identity Switcher" yet, most people rely on the Conditional Includes in their global git config for local work, but for the web UI, that manual dropdown selection is currently the only official way to toggle it.
```

This didn't prevent me from accidentally Squash Merge with my personal email within my organization.
